core/assertion.ts

```diff
@@ -62,10 +63,18 @@ export class Assertion {
   private contextableQuery: AContextable<string>;
 
   public config(config: IAssertionConfig) {
-    checkExcessProperties(config, IAssertionConfigProperties, "assertion config");
+    checkExcessProperties(
+      (e: Error) => this.session.compileError(e),
```

Comment:
Don't love this, is it also necessary? Each definition require() call basically does this wrapping AIUI, why not just throw?
Or, `try { ... } catch (e) { compileError(e) }`?

Reply:
Would be good to understand intention here!

Reply:
The problem with just throwing is that it short circuits the function, preventing any further code from running. This makes error messages really confusing: the compiler complains that `hermetic` is missing, when it really isn't.

I could move this check to the bottom of the function, or try/catch (though that seems messy when it's invoked in several places across the codebase).

However, IMO because `core` is essentially a transpiler, it should strongly prefer to output as many compile errors as it can; given that, I don't think we should be throwing uncaught Errors basically anywhere (unless the following code really cannot continue to run).

Reply:
thank you, I understand now!